GeometryCore/src/Mesh.cpp

```diff
@@ -164,4 +165,47 @@ namespace geom_utils
 		return true;
 	}
 
+	Mesh createFoundation(const Mesh& inputModel, const FPoint3D::coord foundationHeight, const FPoint3D::coord inflateValue) {
+		auto from2dTo3dTriangle = [](const Triangle2D& inputTriangle) { return Triangle3D(FPoint3D(inputTriangle.a.x, inputTriangle.a.y, 0), FPoint3D(inputTriangle.b.x, inputTriangle.b.y, 0), FPoint3D(inputTriangle.c.x, inputTriangle.c.y, 0)); };
+		auto moveUp = [](std::vector<Triangle3D>& foundationPart, FPoint3D::coord by) {for_each(foundationPart.begin(), foundationPart.end(), [&](Triangle3D& input) {input.a.z += by; input.b.z += by; input.c.z += by; }); };
+
+		std::vector<Triangle3D> facets(inputModel.getFacets());
+		Polygon verticles;
+
+		for (auto i : facets) {
+			verticles.add(FPoint2D(i.a.x, i.a.y));
+			verticles.add(FPoint2D(i.b.x, i.b.y));
+			verticles.add(FPoint2D(i.c.x, i.c.y));
+		}
+		auto convex = verticles.convexHull();
+		convex.simplify(convex.polygonLength() * 0.01);
+		auto planeForFoundation = convex.inflate(inflateValue);
+
+		auto polygonOfBottom = planeForFoundation.triangulate();
+
+		std::vector<Triangle3D> foundationBottom;
+		std::transform(polygonOfBottom.begin(), polygonOfBottom.end(), std::back_inserter(foundationBottom), from2dTo3dTriangle);
+		auto foundationTop(foundationBottom);
+		moveUp(foundationTop, foundationHeight);
+
+		Mesh foundation;
+		std::transform(polygonOfBottom.begin(), polygonOfBottom.end(), std::back_inserter(foundationBottom), &Triangle3D::reverse);
```

Comment:
I suggest you to review this line one more time. It doesnt work in the way you may expect. 
Please read documentation with examples about back_inserter and about using of class methods as predicates.

Also, never commit/push code changes if you didn't check them firstly whether changes work as they should. Unless you have enough experience for committing code without checking it.
@IlliaHryshchuk @okrdima please notice as well

Reply:
sorry, it was my inattention, i forgot to build before run tests

Reply:
@VladGribinchuk , of course, I understand it, checking is always mandatory before committing